### PR TITLE
Reorganize repo: grouped config files inside spyderlib/config

### PR DESCRIPTION
**Pull request by goanpeca**
**Repo organization** (related to issue #1320)

All dangling  `****config.py` files were grouped into a single folder, `spyderlib/config/`

### CODE REVIEW

**Comment by ccordoba12** [comment]
This one needs a rebase :-)

**Comment by goanpeca** [comment]
@ccordoba12 done!

File: spyderlib/utils/inspector/conf.py [hunk 1]
```diff
@@ -10,7 +10,7 @@
 from sphinx import __version__ as sphinx_version
 
 # Local imports
-from spyderlib.config import CONF
+from spyderlib.config.config import CONF
```
Comment (ccordoba12) [inline spyderlib/utils/inspector/conf.py]: This an error, right? It should be

    from spyderlib.config.main import CONF

Comment (goanpeca) [inline spyderlib/utils/inspector/conf.py]: Oops, good catch,...

File: spyderlib/widgets/externalshell/sitecustomize.py [hunk 2]
```diff
@@ -454,7 +454,7 @@ def set_spyder_breakpoints(self):
         bdb.Breakpoint.bplist = {}
         bdb.Breakpoint.bpbynumber = [None]
         #------
-        from spyderlib.config import CONF
+        from spyderlib.config.config import CONF
```
Comment (ccordoba12) [inline spyderlib/widgets/externalshell/sitecustomize.py]: Another error here (the same as above :-)

File: spyderlib/widgets/externalshell/start_ipython_kernel.py [hunk 3]
```diff
@@ -25,7 +25,7 @@ def kernel_config():
     """Create a config object with IPython kernel options"""
     from IPython.config.loader import Config, load_pyconfig_files
     from IPython.core.application import get_ipython_dir
-    from spyderlib.config import CONF
+    from spyderlib.config.config import CONF
```
Comment (ccordoba12) [inline spyderlib/widgets/externalshell/start_ipython_kernel.py]: Same error

File: spyderlib/widgets/ipython.py [hunk 4]
```diff
@@ -33,10 +33,10 @@
...
 # Local imports
-from spyderlib.baseconfig import (get_conf_path, get_image_path,
-                                  get_module_source_path, _)
-from spyderlib.config import CONF
-from spyderlib.guiconfig import (create_shortcut, get_font, get_shortcut,
+from spyderlib.config.base import (get_conf_path, get_image_path,
+                                   get_module_source_path, _)
+from spyderlib.config.main import CONF
+from spyderlib.config.gui import (create_shortcut, get_font, get_shortcut,
                                  new_shortcut)
```
Comment (ccordoba12) [inline spyderlib/widgets/ipython.py]: Missing space to the left here to align this line to the parens above :-)

File: spyderlib/plugins/shortcuts.py [hunk 5]
```diff
@@ -21,8 +21,8 @@
 from spyderlib.qt.compat import to_qvariant, from_qvariant
 
 # Local imports
-from spyderlib.baseconfig import _, debug_print
-from spyderlib.guiconfig import (get_shortcut, set_shortcut,
+from spyderlib.config.base import _, debug_print
+from spyderlib.config.gui import (get_shortcut, set_shortcut,
                                  iter_shortcuts, reset_shortcuts)
```
Comment (ccordoba12) [inline spyderlib/plugins/shortcuts.py]: Missing space to the right, here too :-)

**Comment by ccordoba12** [comment]
I really don't understand how our tests are passing with the errors I pointed out.

How can that be the case @goanpeca?

**Comment by goanpeca** [comment]
The imports were located in if statements... so they were not being triggered...and the one in the inspector was a sphinx ... so there I do not really know.

**Comment by ccordoba12** [comment]
I did a last review, and everything seems in order now :-) Weird that our tests didn't fail...

**Pull request merged**